### PR TITLE
ENH: adding name as a karg in Flight class

### PR DESCRIPTION
**Pull request by Gui-FernandesBR**
## Pull request type

Please check the type of change your PR introduces:

- [x] Code base additions (bugfix, features)
- [ ] Code maintenance (refactoring, formatting, renaming, tests)
- [ ] ReadMe, Docs and GitHub maintenance
- [ ] Other (please describe):

## Pull request checklist

Please check if your PR fulfills the following requirements, depending on the type of PR:

- Code base additions (for bug fixes / features):

  - [ ] Tests for the changes have been added
  - [x] Docs have been reviewed and added / updated if needed
  - [x] Lint (`black rocketpy`) has passed locally and any fixes were made
  - [ ] All tests (`pytest --runslow`) have passed locally

## What is the current behavior?
I cannot easily differentiate one flight object from other.

## What is the new behavior?
New argument and attribute "name" added to Flight class, allowing if to easily label/name your object. It was inspired in the recent AeroSurfaces implementation. This will help a lot on compare plots functions.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information
Could apply to other classes as well, if evrybody thinks

### CODE REVIEW

**Comment by Gui-FernandesBR** [comment]
Flight class now receives 'name' as argument, so you can give a name for your flight and I can do `TestFlight.name` without problem.

1. Does it make sense?
2. Is that good approach?
3. Do you know any better approach?

File: rocketpy/Flight.py [hunk 1]
```diff
@@ -527,6 +529,7 @@ def __init__(
         atol=6 * [1e-3] + 4 * [1e-6] + 3 * [1e-3],
         timeOvershoot=True,
         verbose=False,
+        name="",
```
Comment (MateusStano) [inline rocketpy/Flight.py:532]: To make it more standardized with what is in AeroSurfaces I think the default for the `name` parameter should be `name = "Flight"`

**Comment by MateusStano** [comment]
> Flight class now receives 'name' as argument, so you can give a name for your flight and I can do `TestFlight.name` without problem.
> 
> 1. Does it make sense?
> 2. Is that good approach?
> 3. Do you know any better approach?

1. yes
2. i think so
3. no

I just don't like how many things we are adding to develop branch when all additions that aren't bug fixes should be going to beta/v1.0. Every addition to develop is another merge/rebase we have to do in beta/v1.0

File: rocketpy/Flight.py (hunk 1 above)
Comment (Gui-FernandesBR) [inline rocketpy/Flight.py:532]:
```suggestion
        name="Flight",
```

File: rocketpy/Flight.py [hunk 2]
```diff
@@ -582,6 +585,8 @@ def __init__(
             time in some cases. Default is True.
         verbose : bool, optional
             If true, verbose mode is activated. Default is False.
+        name : str, optional
+            Name of the flight. Default is "".
```
Comment (Gui-FernandesBR) [inline rocketpy/Flight.py:589]:
```suggestion
            Name of the flight. Default is "Flight".
```

**Comment by Gui-FernandesBR** [comment]
> > Flight class now receives 'name' as argument, so you can give a name for your flight and I can do `TestFlight.name` without problem.
> > 
> > 1. Does it make sense?
> > 2. Is that good approach?
> > 3. Do you know any better approach?
> 
> 1. yes
> 2. i think so
> 3. no
> 
> I just don't like how many things we are adding to develop branch when all additions that aren't bug fixes should be going to beta/v1.0. Every addition to develop is another merge/rebase we have to do in beta/v1.0

I've updated the PR then, thank you for reviewing

**Comment by MateusStano** [comment]
Seems good to me then. Feel free to merge

**Pull request merged**